operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -0,0 +1,150 @@
+// Package v1alpha1 is an API version in the controlplane.operator.openshift.io group
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodNetworkConnectivityCheck
+type PodNetworkConnectivityCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the source and target of the connetivity check
+	// +kubebuilder:validation:Required
+	// +required
+	Spec PodNetworkConnectivityCheckSpec `json:"spec"`
+
+	// Status contains the observed status of the connectivity check
+	// +optional
+	Status PodNetworkConnectivityCheckStatus `json:"status"`
+}
+
+type PodNetworkConnectivityCheckSpec struct {
+	// SourcePod names the pod from which the condition will be checked
+	// +kubebuilder:validation:Required
+	// +required
+	SourcePod string `json:"sourcePod"`
```

Comment:
This is really just documentation, isn't it? It won't magically add another tester sidecar. Would document that here.

Reply:
1. The sidecar is added by the operator.
2. The `PodNetworkConnectivityCheck` resources also created by operator: one **per** pod, **per** endpoint.
3. The sidecar uses this field to find the endpoints it needs to test, and updates the status with results.

Reply:
> This is really just documentation, isn't it? It won't magically add another tester sidecar. Would document that here.

the container looks up it's own pod name to figure out what to try to contact

Reply:
oh, ok. So in theory one could change that. Just a new object won't have any effect without a sidecar being deployed for a matching pod.